fastfold/workflow/template/fastfold_data_workflow.py

```diff
@@ -89,66 +143,33 @@ def run(self, fasta_path: str, output_dir: str, alignment_dir: str=None) -> None
                 os.makedirs(alignment_dir)
 
         # Run JackHmmer on UNIREF90
-        # create JackHmmer workflow generator
-        jh_config = {
-            "binary_path": self.db_map["jackhmmer"]["binary"],
-            "database_path": self.db_map["jackhmmer"]["dbs"][0],
-            "n_cpu": self.no_cpus,
-            "uniref_max_hits": self.uniref_max_hits,
-        }
-        jh_fac = JackHmmerFactory(config = jh_config)
-        # set jackhmmer output path
         uniref90_out_path = os.path.join(alignment_dir, "uniref90_hits.a3m")
         # generate the workflow with i/o path
-        jh_node_1 = jh_fac.gen_node(fasta_path, uniref90_out_path)
+        uniref90_node = self.jackhmmer_uniref90_factory.gen_node(fasta_path, uniref90_out_path)
 
         #Run HHSearch on STEP1's result with PDB70"""
-        # create HHSearch workflow generator
-        hhs_config = {
-            "binary_path": self.db_map["hhsearch"]["binary"],
-            "databases": self.db_map["hhsearch"]["dbs"],
-            "n_cpu": self.no_cpus,
-        }
-        hhs_fac = HHSearchFactory(config=hhs_config)
-        # set HHSearch output path
         pdb70_out_path = os.path.join(alignment_dir, "pdb70_hits.hhr")
         # generate the workflow (STEP2 depend on STEP1)
-        hhs_node = hhs_fac.gen_node(uniref90_out_path, pdb70_out_path, after=[jh_node_1])
+        hhs_node = self.hhsearch_pdb_factory.gen_node(uniref90_out_path, pdb70_out_path, after=[uniref90_node])
 
         # Run JackHmmer on MGNIFY
-        # reconfigure jackhmmer factory to use MGNIFY DB instead
-        jh_fac.configure('database_path', self.db_map["jackhmmer"]["dbs"][1])
-        # set jackhmmer output path
         mgnify_out_path = os.path.join(alignment_dir, "mgnify_hits.a3m")
         # generate workflow for STEP3
-        jh_node_2 = jh_fac.gen_node(fasta_path, mgnify_out_path)
+        mgnify_node = self.jackhmmer_mgnify_factory.gen_node(fasta_path, mgnify_out_path)
 
         if not self.use_small_bfd:
             # Run HHBlits on BFD
-            # create HHBlits workflow generator
-            hhb_config = {
-                "binary_path": self.db_map["hhblits"]["binary"],
-                "databases": self.db_map["hhblits"]["dbs"],
-                "n_cpu": self.no_cpus,
-            }
-            hhb_fac = HHBlitsFactory(config=hhb_config)
-            # set HHBlits output path
             bfd_out_path = os.path.join(alignment_dir, "bfd_uniclust_hits.a3m")
             # generate workflow for STEP4
-            hhb_node = hhb_fac.gen_node(fasta_path, bfd_out_path)
+            bdf_node = self.hhblits_bfd_factory.gen_node(fasta_path, bfd_out_path)
```

Comment:
bfd_node, the same below

Reply:
it is ok, there is only one bdf_node

Reply:
it depend on if used small bdf

Reply:
no, I said typo. we have nothing to do with `bdf`, we are using `bfd` database.